Conch/lib/Conch/ValidationSystem.pm

```diff
@@ -71,95 +70,76 @@ sub load_validations ( $class, $logger = Mojo::Log->new ) {
+associating validation plans with devices through Orchestration is available to
+users.
 
-	my $validation_plan =
-		Conch::Model::ValidationPlan->lookup($validation_plan_id);
-	Mojo::Exception->throw(
-		"No Validation Plan found with ID '$validation_plan_id'")
-		unless $validation_plan;
-
-	my $validations = $validation_plan->validation_ids;
-	Mojo::Exception->throw(
-		"Validation Plan $validation_plan_id is not associated with any validations"
-	) unless scalar( $validations->@* );
-
-	my $validation_state =
-		Conch::Model::ValidationState->create( $device_id, $validation_plan->id );
-
-	my $minion             = Conch::Minion->new;
-	my @validation_job_ids = map {
-		$minion->enqueue(
-			validation => [ $_, $device_id, $data, $validation_state->id ] );
-	} $validations->@*;
+=cut
 
-	$minion->enqueue(
-		commit_validation_state => [ $validation_state->id ],
-		{ parents => \@validation_job_ids }
-	);
+sub load_legacy_plans ( $class, $logger = Mojo::Log->new ) {
 
-	return $validation_state;
-}
+	my $switch_plan =
+		Conch::Model::ValidationPlan->lookup_by_name(
+		'Conch v1 Legacy Plan: Switch');
 
-=head2 start_tasks
+	unless ($switch_plan) {
+		$switch_plan = Conch::Model::ValidationPlan->create(
+			'Conch v1 Legacy Plan: Switch',
+			'Validation plan containing all validations run in Conch v1 on switches'
+		);
+		$logger->debug( "Created validation plan " . $switch_plan->name );
+	}
 
-Start the Minion tasks for processing validations
+	my @switch_validations =
+		qw( bios_firmware_version cpu_count cpu_temperature product_name
+		dimm_count ram_total );
+	for my $name (@switch_validations) {
+		my $validation =
+			Conch::Model::Validation->lookup_by_name_and_version( $name, 1 );
+		if ($validation) {
+			$switch_plan->add_validation($validation);
+		}
+		else {
+			$logger->warn( "Could not find Validation name $name, version 1"
+					. " to load for Legacy Switch Validation Plan" );
+		}
+	}
 
-=cut
+	my $server_plan =
+		Conch::Model::ValidationPlan->lookup_by_name(
+		'Conch v1 Legacy Plan: Server');
+	unless ($server_plan) {
+		$server_plan =
+			Conch::Model::ValidationPlan->create( 'Conch v1 Legacy Plan: Server',
+			'Validation plan containing all validations run in Conch v1 on servers' );
+		$logger->debug( "Created validation plan " . $server_plan->name );
+	}
 
-sub start_tasks ( $class ) {
-	Conch::Minion->new->add_task(
-		validation => sub {
-			my ( $job, $validation_id, $device_id, $data ) = @_;
-
-			my $validation = Conch::Model::Validation->lookup($validation_id);
-			$job->fail("Unable to find Validation '$validation_id'")
-				unless $validation;
-			my $device = Conch::Model::Device->lookup($device_id);
-			$job->fail("Unable to find Device '$device_id'")
-				unless $device;
-
-			my @results = map { $_->output_hash }
-				$validation->run_validation_for_device( $device, $data )->@*;
-			$job->finish( \@results );
+	my @server_validations = qw( bios_firmware_version cpu_count cpu_temperature
```

Comment:
Should this maybe be out in a config someplace?

Reply:
I'll do this. I initially decided against it because I hope to drop these legacy validation plans soon as we move to the new orchestration/validation world, but it's entirely likely that we might add new validations to these plans before we fully migrate over.